src/QuantumSdk/DefaultItems/DefaultItems.targets

```diff
@@ -107,17 +115,24 @@
       <_ResolvedTargetPackageReference Include="$(%(_TargetPackageReferencePathProperty.Identity))" />
       <ResolvedTargetSpecificDecompositions Include="@(_ResolvedTargetPackageReference->Split(`;`))" />
     </ItemGroup>
-    <Error Condition="@(ResolvedTargetSpecificDecompositions->Count()) &gt; 1" Text="More than one target package was defined. The defined target package needs to be unique." />
     <Error
       Condition="@(ResolvedTargetSpecificDecompositions->Count()) == 0 And @(_ResolvedTargetPackageReference->Count()) &gt; 0"
       Text="Invalid target package. The target package is expected to specify the dll containing the decompositions to load." />
     <Message
       Condition="@(ResolvedTargetSpecificDecompositions->Count()) &gt; 0 And ('$(QscVerbosity)' == 'Detailed' Or '$(QscVerbosity)' == 'Diagnostic')"
-      Text="Resolved target package assembly: @(ResolvedTargetSpecificDecompositions)"
+      Text="Resolved target package assemblies: @(ResolvedTargetSpecificDecompositions)"
       Importance="High" />
   </Target>
 
-  <!-- Computes ResolvedQscReferences, ResolvedQscReferencesAndPriorities, QscProjectReference and QscPackageReference. -->
+  <!-- Populates the ResolvedQSharpReferences item group. -->
+  <Target Name="ResolveQSharpReferences" DependsOnTargets="ResolveTargetPackage;ResolveAssemblyReferences;_CopyFilesMarkedCopyLocal" BeforeTargets="BeforeQSharpCompile">
+    <ItemGroup>
+      <ResolvedQSharpReferences Include="@(ReferencePath)" Condition="$([System.Text.RegularExpressions.Regex]::IsMatch(%(FullPath), '(?i)system.|mscorlib|netstandard.library|microsoft.netcore.app|csharp|fsharp|microsoft.visualstudio|microsoft.testplatform|microsoft.codeanalysis|fparsec|newtonsoft|roslynwrapper|yamldotnet|markdig|serilog')) == false" />
+      <ResolvedQSharpReferences Exclude="@(ResolvedTargetSpecificDecompositions)" />
```

Comment:
I opted that both the core package to support compilation in the first place as well as the target specific decompositions should be referenced by the project. That means that the target package can contain multiple dlls, and some or all of these dlls may be declared as the actual target decompositions. The content of dlls that are not explicitly declared as target specific decompositions in the target package are exposed in exactly the same way that other references are and will hence be visible to the user, while the dll content of target specific decompositions is only loaded after the initial compilation such that it does not conflict with other declarations and instead replaces them. Declarations in those dlls are not visible to the user.